presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -14,15 +14,176 @@
 package io.prestosql.plugin.jdbc;
 
 import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import io.airlift.units.Duration;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.SchemaTableName;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
 import java.util.Set;
 
+import static com.google.common.collect.Sets.difference;
 import static io.prestosql.spi.testing.InterfaceTestUtils.assertAllMethodsOverridden;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static org.assertj.core.api.Assertions.assertThat;
 
+@Test(singleThreaded = true)
 public class TestCachingJdbcClient
 {
+    private static final Duration FOREVER = Duration.succinctDuration(1, DAYS);
+    private static final ConnectorSession session = testSessionBuilder().build().toConnectorSession();
+    private static final JdbcIdentity identity = JdbcIdentity.from(session);
+
+    private TestingDatabase database;
+    private CachingJdbcClient cachingJdbcClient;
+    private JdbcClient dbHacker;
+    private String schema;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        database = new TestingDatabase();
```

Comment:
Why not to reuse it between tests? Then you can use `io.prestosql.testing.AbstractTestQueryFramework#closeAfterClass`. Then you can make it concurrent again